### ISSUE TITLE
Makefile install_all should not install test binary

### ISSUE THREAD

**Issue by J08nY** [libtom/libtomcrypt#396]
### Prerequisites

* [x] Checked the developer manual
* [x] Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=repo%3Alibtom%2Flibtomcrypt
* [x] Checked that your issue isn't related to TomsFastMath's limitation that PK operations can by default only be done with max. 2048bit keys

### Description

`make -f makefile.shared install_all` installs the `test` binary into $PATH, which then shadows coreutils `test` binary and causes problems. Generally installing a binary named `test` will cause some issues.

### Steps to Reproduce

`make -f makefile.shared install_all`

### Version

1.18.1

**Comment by sjaeckel** [comment]
Isn't that kind of a 'feature' to have a good replacement in case your regular `test` is defect? ;) \</irony>

**Comment by sjaeckel** [comment]
Jokes aside, we have the following set of demos etc.

```
aesgcm 2*
constants 1*
hashsum 2*
ltcrypt 2*
openssl-enc 2*
sizes 1*
small 0*
test 0*
timing 0*
tv_gen 0*
```

I've marked my opinion on their "install-worthiness"

As we had a nameclash for `crypt` which led us to rename it to `ltcrypt` we have the same for `hashsum` which we didn't resolve yet...

I'd say we should remove the possibility to install all "0*-rated" demos and the 1*- and 2*-rated demos should be renamed to something obvious e.g. `ltcsizes` and `ltconstants`. Then also regroup them... or don't use groups at all, but provide install-targets per binary?

If we keep groups:
```
demos - Builds 2*
bins - Builds 1* and 2*
all - Builds library, 0*, 1* and 2*
install_demos - Build&Install 2*
install_bins - Build&Install 1* and 2*
install_all - install install_bins install_doc (as before)
```

Please add your opinions & preferences! (ping @karel-m @buggywhip)

**Comment by J08nY** [comment]
I agree with all of the above, also the renaming/prefixing with `ltc` is useful and should allow easy installing of those binaries into $PATH without name clashes. I also think grouping in Makefiles is useful and allows someone to easily play with the makefile so the demos should be regrouped.

Anyway, thanks for working on this issue! :+1:

**Comment by karel-m** [comment]
IMO the only tool in "release quality" is `hashsum` and it is the only candidate worth installing. I am in doubts whether anybody will ever use `make install_demos`.

The `ltcrypt` + `openssl-enc` completely ignore integrity and `aesgcm` IMO needs a bit more work (we have discussed it in some other PR).

The prefixing with `ltc` is also not a bad idea.

**Comment by sjaeckel** [comment]
I'm reopening this so we can re-work as proposed.
